nsis/SectionVersionCheck.nsh

```diff
@@ -0,0 +1,98 @@
+; Copyright (C) 2017 Christian Kenny
+
+Function UninstallPrevious
+
+; Here's how StrContains works:
+; $result_var: This will store our result if the "needle" is found. Otherwise it will return null ("")
+; $needle: String to search for
+; $Haystack: String to look in
+
+; Search for 1.0.0
+ReadRegStr $R7 HKLM "${INSTDIR_REG_KEY}" "Uninst-pcsx2Directory"
+${StrContains} "$4" "1.0.0" "$R7"
+
+; This will become the primary version check for 1.6.0 and later
+ReadRegStr $R1 HKLM "${INSTDIR_REG_KEY}" "DisplayVersion"
+
+; Search for 1.4.0
+ReadRegStr $R2 HKLM "${INSTDIR_REG_KEY}" "Uninst-pcsx2 1.4.0Directory"
+${StrContains} "$2" "1.4.0" "$R2"
+
+; Search for 1.2.1
+ReadRegStr $R3 HKLM "${INSTDIR_REG_KEY}-r5875" "Uninst-pcsx2-r5875Directory"
+${StrContains} "$3" "1.2.1" "$R3"
+
+; Search for 0.9.8
+ReadRegStr $R5 HKLM "${INSTDIR_REG_KEY}-r4600" "Uninst-pcsx2-r4600Directory"
+${StrContains} "$5" "0.9.8" "$R5"
+
+; If all cases return null, our work here is done.
+${If} $R2 == ""
+${AndIf} $R3 == ""
+${AndIf} $R5 == ""
+${AndIf} $R7 == ""
+Return
+${EndIf}
+
+; Installing another version
+MessageBox MB_ICONEXCLAMATION|MB_OKCANCEL "Another version of PCSX2 is already installed. The current configuration folder in Documents will be duplicated and renamed as PCSX2_backup. Click OK to uninstall PCSX2 or Cancel to abort the setup." IDOK SetUninstPath IDCANCEL false
+
+false:
```

Comment:
Not sure you understood me. If the returned MessageBox button ID doesn't match any of the return checks then execution will resume at the next line. So the `IDCANCEL false` part of the MessageBox line and the `false:` label are unnecessary.

Note: This isn't really a big issue. You can leave it if you want. And it's also based on my reading of the NSIS documentation - I haven't actually used NSIS before (not sure if I mentioned it before), so I could be misunderstanding something.